ra/ra.go

```diff
@@ -373,6 +373,7 @@ func (ra *RegistrationAuthorityImpl) NewAuthorization(ctx context.Context, reque
 				fmt.Sprintf("unable to get existing validations for regID: %d, identifier: %s",
 					regID, identifier.Value))
 			ra.log.Warning(string(outErr))
+			return authz, outErr
```

Comment:
It looks like there's another example of this failure to return an error in the next if stanza, below. You should fix that as part of this PR.